dev/com.ibm.ws.concurrent.persistent_fat_multiple/fat/src/com/ibm/ws/concurrent/persistent/fat/multiple/DatabaseContainerFactory.java

```diff
@@ -0,0 +1,144 @@
+package com.ibm.ws.concurrent.persistent.fat.multiple;
+
+import java.time.Duration;
+
+import org.testcontainers.containers.Db2Container;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.MockServerContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.MSSQLServerContainer;
+import org.testcontainers.containers.OracleContainer;
+import org.testcontainers.containers.output.OutputFrame;
+
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.custom.junit.runner.FATRunner;
+
+/**
+ * This is a factory class that creates a database test container.
+ * The test container returned will be based on the 'fat.bucket.db.type' 
+ * system property. 
+ * 
+ * The 'fat.bucket.db.type' property is set to different databases
+ * by our test infrastructure when a fat-suite has the property
+ * 'fat.test.databases' set to true
+ * 
+ * TODO: Since this factory class is general can it be moved to fattest simplicity?
+ *
+ */
+public class DatabaseContainerFactory {
+	private static String database = System.getProperty("fat.bucket.db.type", "Derby");
```

Comment:
instead of using a String here and in the switch blocks, perhaps we could make/reuse an enum for the DBs supported? This would help enforce more desirable behavior for an unknown/misspelled DB type raising an exception instead of silently falling back to using Derby